src/module_3/push_mvp.py

```diff
@@ -0,0 +1,121 @@
+
+
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import  StandardScaler
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.compose import ColumnTransformer
+import category_encoders as ce
+from sklearn.model_selection import StratifiedKFold
+import matplotlib.pyplot as plt
+from datetime import timedelta
+from sklearn.model_selection import PredefinedSplit
+import pyarrow.parquet as pq
+from sklearn import metrics
+from sklearn.metrics import roc_auc_score, confusion_matrix
+from evaluation_functions import *
+
+file_path = '/home/oscar/data/feature_frame.parquet'
+save_models_path = 'src/module_3/models'
+
+
+def read_data(file_path: str) -> pd.DataFrame:
+    try:
+        data = pq.read_table(file_path).to_pandas()
+        
+        print("Data shape:", data.shape)
+        print("Data columns:", data.columns.tolist())
+        print("Data types:\n", data.dtypes)
+        
+        return data
+    except FileNotFoundError:
+        print("Error: File not found at the specified path.")
+        return None
+    except Exception as e:
+        print("An error occurred:", e)
+        return None
+
+def sample_data_by_orders(data: pd.DataFrame, sample_percentage: float) -> pd.DataFrame:
+
+    order_ids = data['order_id'].unique()
+    
+    sample_orders_amount = int(len(order_ids) * sample_percentage)
+    
+    sampled_orders = np.random.choice(order_ids, size=sample_orders_amount, replace=False)
+    
+    sampled_data = data[data['order_id'].isin(sampled_orders)]
+    
+    print("Original dataset size:", data.shape)
+    print("Sampled dataset size:", sampled_data.shape)
+    
+    return sampled_data
+
+def preprocess_data(data: pd.DataFrame, num_items_to_filter: int, remove_na: bool, features_to_remove: list) -> pd.DataFrame:
+
+    num_products_ordered = data.groupby('order_id')['outcome'].sum()
+    
+    orders_with_min_num_products = num_products_ordered[num_products_ordered >= num_items_to_filter].index
+    filtered_data = data[data['order_id'].isin(orders_with_min_num_products)]
+    
+    if remove_na:
+        filtered_data = filtered_data.dropna()
```

Comment:
I'd be very careful with this type of operations. You could be dropping most of your dataset without noticing if a column starts having nans for whatever reason.